lms/views/api/canvas/sync.py

```diff
@@ -38,12 +38,20 @@ def group(section):
         hash_object.update(str(section["id"]).encode())
         authority_provided_id = hash_object.hexdigest()
 
-        return HGroup(
-            h_group_name(section["name"]), authority_provided_id, type="section_group"
-        )
+        if "name" in section:
+            name = h_group_name(section["name"])
+        else:
+            name = None
 
-    groups = [group(section) for section in sections]
+        return HGroup(name, authority_provided_id, type="section_group")
 
+    groups = [group(section) for section in sections]
     lti_h_svc.sync(groups, group_info)
 
+    if "learner" in request.json:
+        user_id = request.json["learner"]["canvas_user_id"]
```

Comment:
I'm a little confused about the structure here. In the case of a teacher in a SpeedGrader context we get the teachers sections and sync them first, then get the users (don't sync them) and return.

Presumably we don't need to sync the students as the teachers are a superset. But do we need to do the teachers at all? Could we just get the students and sync them? This would cut a call to Canvas, which could be significant as this happens every time you skip to a new student while grading.